src/dtx/dtx_rpc.c

```diff
@@ -446,14 +452,14 @@ dtx_dti_classify_one(struct ds_pool *pool, struct pl_map *map, uuid_t po_uuid,
 	dcrb.dcrb_length = length;
 
 	crt_group_rank(pool->sp_group, &myrank);
-	start = (oid->id_shard / replicas) * replicas;
-	for (i = start; i < start + replicas && rc >= 0; i++) {
+	start = (oid->id_shard / tgt_cnt) * tgt_cnt;
+	for (i = start; i < start + tgt_cnt && rc >= 0; i++) {
```

Comment:
it is fine for full stripe update. But for partial update a problem is that we need to find a way to only send the RPC to the update involved tgts rather than all members in the group if possible.

Reply:
Yes, I plan to look at storing a bit-map with the entry for the batched commits.  However, it seems that sending an RPC to slave targets that are not involved in the update does not cause an issue. Definitely want to fix this though.

Reply:
> Yes, I plan to look at storing a bit-map with the entry for the batched commits. However, it seems that sending an RPC to slave targets that are not involved in the update does not cause an issue. Definitely want to fix this though.


If you send DTX_COMMIT/DTX_ABORT RPC to the target that is not involved in the update, it will not cause failure, just some fake warning message. But if you send DTX_CHECK RPC to such target, it will misguide the RPC sponsor to abort related DTX by wrong. That should be avoided.